### PR TITLE
Add a "hello-world" example with TLS configured.

### PR DESCRIPTION
**Pull request by nddipiazza**
Add a hello-world-client-tls and hello-world-server-tls examples of setting up grpc-jaa with TLS

see https://github.com/grpc/grpc-java/pull/3992/files#diff-1c0f522a61adc59307209c8e0296db49 for setup and usage

### CODE REVIEW

**Comment by thelinuxfoundation** [comment]
Thank you for your pull request. Before we can look at your contribution, we need to ensure all contributors are covered by a Contributor License Agreement.

After the following items are addressed, please respond with a new comment here, and the automated system will re-verify.

* User @nddipiazza isn't covered by a CLA. They will need to complete the form at https://identity.linuxfoundation.org/projects/cncf

_Regards,
CLA GitHub bot_

**Comment by nddipiazza** [comment]
@ejona86 @carl-mastrangelo this is ready for review.

File: examples/src/main/java/io/grpc/examples/helloworldtls/HelloWorldClientTls.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+/**
+ * A simple client that requests a greeting from the {@link HelloWorldServer} with TLS.
+ */
+public class HelloWorldClientTls {
+  private static final Logger logger = Logger.getLogger(HelloWorldClientTls.class.getName());
+
+  private final ManagedChannel channel;
+  private final GreeterGrpc.GreeterBlockingStub blockingStub;
+
+  private static SslContext buildSslContext(String trustCertCollectionFilePath,
```
Comment (carl-mastrangelo) [inline examples/src/main/java/io/grpc/examples/helloworldtls/HelloWorldClientTls.java]: optional:  we typically use +4 indent for line wrap

File: examples/src/main/java/io/grpc/examples/helloworldtls/HelloWorldClientTls.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+   * Say hello to server.
+   */
+  public void greet(String name) {
+    logger.info("Will try to greet " + name + " ...");
+    HelloRequest request = HelloRequest.newBuilder().setName(name).build();
+    HelloReply response;
+    try {
+      response = blockingStub.sayHello(request);
+    } catch (StatusRuntimeException e) {
+      logger.log(Level.WARNING, "RPC failed: {0}", e.getStatus());
```
Comment (carl-mastrangelo) [inline examples/src/main/java/io/grpc/examples/helloworldtls/HelloWorldClientTls.java]: consider letting the exception propagate.

File: examples/src/main/java/io/grpc/examples/helloworldtls/HelloWorldClientTls.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+      System.out.println("USAGE: HelloWorldClientTls host port [trustCertCollectionFilePath] [clientCertChainFilePath] " +
+          "[clientPrivateKeyFilePath]\n  Note: clientCertChainFilePath and clientPrivateKeyFilePath are only needed " +
+          "if mutual auth is desired. And if you specify clientCertChainFilePath you must also specify " +
+          "clientPrivateKeyFilePath");
+      System.exit(0);
+    }
+
+    HelloWorldClientTls client = new HelloWorldClientTls(args[0],
+        Integer.parseInt(args[1]),
+        buildSslContext(args.length > 2 ? args[2] : null, args.length > 3 ? args[3] : null, args.length > 4 ? args[4] : null));
```
Comment (carl-mastrangelo) [inline examples/src/main/java/io/grpc/examples/helloworldtls/HelloWorldClientTls.java]: I personally find this hard to read.   Maybe consider just switching on args.length, and constructing `client`  from the appropriate params?

File: examples/src/main/java/io/grpc/examples/helloworldtls/HelloWorldClientTls.java (hunk 2 above)
Comment (nddipiazza) [reply examples/src/main/java/io/grpc/examples/helloworldtls/HelloWorldClientTls.java]: this code i didn't write. it's copied from hello world existing

File: examples/src/main/java/io/grpc/examples/helloworldtls/HelloWorldClientTls.java (hunk 1 above)
Comment (nddipiazza) [reply examples/src/main/java/io/grpc/examples/helloworldtls/HelloWorldClientTls.java]: will do

File: examples/src/main/java/io/grpc/examples/helloworldtls/HelloWorldClientTls.java (hunk 3 above)
Comment (nddipiazza) [reply examples/src/main/java/io/grpc/examples/helloworldtls/HelloWorldClientTls.java]: sure. will do

**Comment by nddipiazza** [comment]
@carl-mastrangelo - i have updated the changes you requested. Except i did not modify the existing code i took from HelloWorld*.java to keep it consistent.

**Comment by nddipiazza** [comment]
@carl-mastrangelo @ejona86 can we merge this before more conflicts arise?

**Comment by ejona86** [comment]
It appears the blaze CI is failing at the very least because of strict dependencies. It shouldn't be too hard to fix. I'll try to take a look, but if you beat me to it, here's the log:
https://source.cloud.google.com/results/invocations/30604bd5-84f0-482d-8927-420c692b9897

**Comment by nddipiazza** [comment]
@ejona86 

Can you check if i fixed it by adding the netty-tcnative dependency to pom.xml (previously i only did build.gradle) and also make sure that putting in the tcnative version 2.0.7.final is ok? we can re-run bazel to see if it works

**Comment by ejona86** [comment]
Yes. 2.0.7 is correct. Bazel doesn't use the pom.xml at all; it uses the BUILD instead.

**Comment by ejona86** [comment]
@nddipiazza, thank you!

**Comment by nddipiazza** [comment]
@ejona86 no, no... thank you!!!

**Comment by sanjaypujare** [comment]
@nddipiazza  I saw your stackoverflow comment regarding mTLS (https://stackoverflow.com/questions/48389074/java-grpc-tls-how-to-set-up-mutual-tls-on-the-client-side). Do you have an mTLS example? If you have some code can you share?

**Comment by nddipiazza** [comment]
@sanjaypujare doesn't the example I added here do exactly that?

**Comment by sanjaypujare** [comment]
@nddipiazza yes, you are right. Sorry I had not looked at the code earlier

**Pull request merged**